File: backend/src/main/java/com/ihecride/utils/AesUtil.java
package com.ihecride.utils;

import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;
import java.security.MessageDigest;
import java.util.Base64;

/**
 * Chiffrement AES-256 (CBC/PKCS5) pour les données sensibles.
 */
public class AesUtil {

    private static final String ALGO = "AES/CBC/PKCS5Padding";
    private static final String SECRET = "IHECRideAesKey2026ChangeInProduction";

    private static SecretKeySpec getKey() throws Exception {
        byte[] key = MessageDigest.getInstance("SHA-256")
                .digest(SECRET.getBytes("UTF-8"));
        return new SecretKeySpec(key, "AES");
    }

    public static String encrypt(String value) {
        try {
            byte[] iv = new byte[16];
            Cipher c = Cipher.getInstance(ALGO);
            c.init(Cipher.ENCRYPT_MODE, getKey(), new IvParameterSpec(iv));
            return Base64.getEncoder().encodeToString(c.doFinal(value.getBytes("UTF-8")));
        } catch (Exception e) {
            throw new RuntimeException("Erreur chiffrement AES", e);
        }
    }

    public static String decrypt(String cipherText) {
        try {
            byte[] iv = new byte[16];
            Cipher c = Cipher.getInstance(ALGO);
            c.init(Cipher.DECRYPT_MODE, getKey(), new IvParameterSpec(iv));
            return new String(c.doFinal(Base64.getDecoder().decode(cipherText)), "UTF-8");
        } catch (Exception e) {
            throw new RuntimeException("Erreur déchiffrement AES", e);
        }
    }
}

File: backend-php/utils/AesUtil.php
<?php
/**
 * Chiffrement AES-256 (CBC/PKCS7) pour les données sensibles.
 * Équivalent PHP de com.ihecride.utils.AesUtil.
 */

class AesUtil {

    private const SECRET = 'IHECRideAesKey2026ChangeInProduction';
    private const ALGO   = 'aes-256-cbc';

    private static function getKey(): string {
        return hash('sha256', self::SECRET, true);   // 32 octets
    }

    public static function encrypt(string $value): string {
        $iv     = str_repeat("\0", 16);              // IV nul (mêmes conditions que le code Java)
        $cipher = openssl_encrypt($value, self::ALGO, self::getKey(), OPENSSL_RAW_DATA, $iv);
        if ($cipher === false) {
            throw new RuntimeException('Erreur chiffrement AES');
        }
        return base64_encode($cipher);
    }

    public static function decrypt(string $cipherText): string {
        $iv    = str_repeat("\0", 16);
        $clear = openssl_decrypt(base64_decode($cipherText), self::ALGO, self::getKey(), OPENSSL_RAW_DATA, $iv);
        if ($clear === false) {
            throw new RuntimeException('Erreur déchiffrement AES');
        }
        return $clear;
    }
}
